src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -896,6 +898,43 @@ public void setHealth(double health) {
                 world.dropItemNaturally(getLocation(), item);
             }
         }
+
+        // TODO: Add a die method to GlowEntity class and override in
+        // various subclasses depending on the actions needed to be run
+        // to help keep code maintainable
+        if (this instanceof GlowSlime) {
+            GlowSlime slime = (GlowSlime) this;
+
+            int size = slime.getSize();
+            if (size > 1) {
+                int count = 2 + ThreadLocalRandom.current().nextInt(3);
+
+                SlimeSplitEvent event = EventFactory.getInstance().callEvent(
+                        new SlimeSplitEvent(slime, count));
+                if (!event.isCancelled() && event.getCount() > 0) {
+                    count = event.getCount();
+                } else {
+                    return;
+                }
+
+                for (int i = 0; i < count; ++i) {
+                    Location spawnLoc = getLocation().clone();
+                    spawnLoc.add(Math.random(), 0, Math.random()); //TODO: Not sure it's vanilla
```

Comment:
Could you use `ThreadLocalRandom.current().nextDouble()` here instead of `Math.random()`?